network/Message.cpp

```diff
@@ -379,34 +342,34 @@ Message GameStartMessage(int player_id, bool single_player_game, int empire_id,
             oa << BOOST_SERIALIZATION_NVP(galaxy_setup_data);
         }
     }
-    return Message(Message::GAME_START, Networking::INVALID_PLAYER_ID, player_id, os.str());
+    return Message(Message::GAME_START, os.str());
 }
 
 Message HostSPAckMessage(int player_id)
-{ return Message(Message::HOST_SP_GAME, Networking::INVALID_PLAYER_ID, player_id, ACKNOWLEDGEMENT); }
+{ return Message(Message::HOST_SP_GAME, std::to_string(player_id)); }
 
 Message HostMPAckMessage(int player_id)
-{ return Message(Message::HOST_MP_GAME, Networking::INVALID_PLAYER_ID, player_id, ACKNOWLEDGEMENT); }
+{ return Message(Message::HOST_MP_GAME, std::to_string(player_id)); }
 
 Message JoinAckMessage(int player_id)
-{ return Message(Message::JOIN_GAME, Networking::INVALID_PLAYER_ID, player_id, ACKNOWLEDGEMENT); }
+{ return Message(Message::JOIN_GAME, std::to_string(player_id)); }
```

Comment:
I don't think using boost::serialization will avoid the chance of a malformed message not being possible to convert into a non-string type, or of an exception being thrown. For cases like this, using it seems like massively over-complicating the implementation.